Makefile

```diff
@@ -1,16 +1,24 @@
+SHELL := /bin/bash
+
+User=${User:=lod}
+
 default: install
 
+all:
+	@echo "Run 'make install' for installation."
+	@echo "Run 'make uninstall' for uninstallation."
+
 requirements:
-	python3 -m pip install -r requirements.txt
+	su ${User} -c "python3 -m pip install -r requirements.txt"
```

Comment:
Deiner Meinung nach ist es nicht besser `su` zu nutzen im Vergleich zu `sudo`. Darüber wollte ich noch kurz sprechen. `sudo` ist ein eigenes Programm, während `su` in den `linux-utils` vorhanden ist und somit auf [fast] jedem Rechner forhanden sein dürfte. Außerderdem ist man bei der Verwendung von `sudo` darauf angewiesen, dass dier lokale benutzer in der `sudo` oder `wheel` Gruppe ist - je nach Konfiguration in der sudoers. Davon kann man nicht ausgehen (z.B. hat der `lod`-Benutzer bei uns keine root-Rechte über `sudo`). Anderherum, bei der Verwendung von `su`, muss man nur sicherstellen, dass `make` von `root` ausgeführt wird. Das geht entweder, indem man sich als root anmeldet oder indem man einen Benutzer nutzt, der selbst sudo-Rechte hat.
Leuchtet das ein?

Reply:
Meiner Meinung nach muss su/sudo/was auch immer komplett aus dem Makefile raus. `make install` sollte dann vom Nutzer mit seinem favorisierten Weg, an root-privilegien zu kommen ausgeführt werden.

Reply:
Was wäre also ein gangbarer Weg, um weder `sudo` noch `su` zu verweden und trotzdem:

1. Das systemd Unit-File an seinen Platz zu bekommen
2. Die Abhängigkeiten der API lokal als nicht-root-Benutzer zu installieren
?